packages/app/resources/views/components/user-menu.blade.php

```diff
@@ -85,6 +101,10 @@
                 <x-filament::dropdown.list.item icon="heroicon-m-sun" x-show="theme === 'light'" x-on:click="close(); mode = 'manual'; theme = 'dark'">
                     {{ __('filament::layout.buttons.dark_mode.label') }}
                 </x-filament::dropdown.list.item>
+
+                <x-filament::dropdown.list.item icon="heroicon-m-cog" x-show="mode === 'manual'" x-on:click="close(); mode = 'auto';">
```

Comment:
Not a huge fan of the UI here. Almost should be a dropdown with "Dark Mode", "Light Mode", "System Default" instead? This would also bring clarity to the verbiage and reduce from 2 buttons down to 1 dropdown?

Reply:
Yeah nested dropdown would be better I think.
Could you take care of that, @bmckay959?
Dropdown component already supports it.

Reply:
I'm actually not sure of best UI here. Let me mockup a few rough options to discuss.

Reply:
Here is the current UI as it stands
<img width="377" alt="Screenshot 2023-04-13 at 11 32 18 AM" src="https://user-images.githubusercontent.com/7681836/231825711-c21aaa65-0625-491d-8361-f0c035801c12.png">

Reply:
I think it should remain in the user menu

Reply:
It should be one item that toggles between the three settings

Reply:
And how would you make that UX clear? 😅

Reply:
Whatever you do, clicking the button should either switch theme or toggle system mode.

If you are currently on dark mode in Filament and dark system, clicking it will set to  light mode.
If you are currently on dark mode in Filament but light system, clicking it will reset to system.
If you are currently on light mode in Filament but dark system, clicking it will reset to system.
If you are currently on light mode in Filament and light system, clicking it will set to dark mode.
If you are currently on light system mode in Filament, clicking it will set to dark mode.
If you are currently on dark system mode in Filament, clicking it will set to light mode.

Reply:
Here is a concept...could smooth out the design if you like the direction? I tend to agree with Zep on the explicit design here feeling more intuitive, but I'll convert what I currently have into what Dan suggested to give it a shot on how it feels.

<img width="394" alt="Screenshot 2023-04-13 at 2 46 21 PM" src="https://user-images.githubusercontent.com/7681836/231867116-dfdf8987-2fda-4559-b74d-ca2af6c9e859.png">

Reply:
Interesting idea, Ben!
I'm worried about mobile/responsiveness with that concept though?

Reply:
Really curious what that UX is like, because it still sounds complicated/confusing to me. 😅

Reply:
I was curious of that too, but essentially, it will always be the opposite of what you currently have.

If you are currently manually toggling themes, the next click makes it auto.
If you are currently on dark, then next click makes it light.

| Current Mode | Current Theme | After Click Mode | After Click Theme |
| -------------- | --------------- | ---------------- | ------------------- |
| Auto | Light | Manual | Dark |
| Manual | Light | Auto | Dark |
| Auto | Dark | Manual | Light |
| Manual | Dark | Auto | Light |

Reply:
But is it clear that it toggles to auto?

Reply:
Just implemented this as an option and pushed.
 Only shows the "Follow System Theme" when mode === manual

<img width="369" alt="Screenshot 2023-04-14 at 9 07 14 AM" src="https://user-images.githubusercontent.com/7681836/232067719-5678ceb8-fa82-42c2-8c58-dc233b972e3e.png">

Reply:
Ok. So, this was a huge learning curve for me, but I think this is a good spot for you guys to review.

The `toggleTheme()` code was easy and self explainatory.

The issue came when a user was on manual or auto, but the system changed themes(Not based on a user click). This is why we have to listen for any theme/mode or systemMode change to handle appropriately.

I actually am really liking the single toggle idea after implementing.

Let me know your thoughts and what I need to change.

Reply:
> it will always be the opposite of what you currently have.

If it will always be the opposite, how is going back to system matters. It feels redundant

Reply:
I don't think having System inside the option list should matter. The user would know whether Light or Dark is their System setting. If we have the choice to choose between light and dark, then system shouldn't be involved here. System mode isn't a thing, light and dark mode is. Adding "System mode" to it just makes it more confusing in my opinion.

Reply:
I disagree. Having the system option allows for automatically matching the system theme when that changes based on time of day. Also good accessibility-wise.

Reply:
I didn't know that your System theme can change on time of day... I think it's fine I was just giving opinion.

Reply:
I think we need @zepfietje  and @danharrin  to decide and get on the same page for how they want it and then I can make a pull request to match that. I see pros and cons to both directions and don't have a strong preference either way.

Reply:
It's quite simple, we should just support light/dark/system.
I'm planning to review this PR as is so we can move forward. 👍

Reply:
Ok. That's not how it is currently working. It is currently just dark/light with auto system toggle if theme != system theme.

I will work on this to get it to the 3 options.

Reply:
Sorry, I didn't necessarily mean 3 options.
Your current implementation already supports light/dark/system.

No work required on your end now.
Will review first and then we can decide if we like it better with 3 options.

Reply:
If this is going to be the decision, in my opinion, I think something like what Facebook does would be the best solution. The layout and placement of it is actually kind of similar to what Filament has now. But I think instead of having all of the options in the first initial click of the dropdown menu, something like this would look cool.

https://user-images.githubusercontent.com/104294090/234752993-7e0b23fa-e2fc-4bb6-a6af-53259d447ddb.mp4

Reply:
Multilevel/Nested Dropdown Menu would be a nice addition as well for Filament V3.

Reply:
> A Multilevel/Nested Dropdown Menu (or whatever the name of this type of dropdown menu is) would be a nice addition as well for Filament V3 for the User Menu.

Yeah that's exactly the solution I would opt for. Little simpler than Facebook, but it does lead to a clearer UX compared to the "magic" system theme toggling with just 1 button.

Reply:
Okay cool! Yeah... the Facebook version would be pretty hard to pull off but would be worth it. Would probably have to require making new dropdown components such as a submenu item with sliding transitions which displays a new dropdown menu list. I couldn't even think of how I would go about doing something like that haha. I'm sure you guys probably might.

Reply:
I don't think it's needed though. A simple nested dropdown would suffice. 🙂

@bmckay959 @danharrin @awcodes could you share your thoughts?
Then we can move this PR forward.

Reply:
As long as the nested dropdown doesn't look weird that sounds good. It should probably open on the left of the main dropdown.

Reply:
We could experiment with that.
Nesting without changing placement didn't look too bad when I tried, but positioning it left is very well possible.

Reply:
Just use the floating ui popup. it's already built in and will handle auto positioning based on available space.